src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -509,10 +509,10 @@ int ControllerScriptInterfaceLegacy::beginTimer(
         return 0;
     }
 
-    if (intervalMillis < 20) {
+    if (intervalMillis < 2) {
         qCWarning(m_logger) << "Timer request for" << intervalMillis
-                            << "ms is too short. Setting to the minimum of 20ms.";
-        intervalMillis = 20;
+                            << "ms is too short. Setting to the (hacked) minimum of 2ms.";
+        intervalMillis = 2;
```

Comment:
This can't be guaranteed on al host operating systems now. And controller mappings must work on all operating systems.

Reply:
FYI Qt 6.8 introduces QChronoTimer https://doc.qt.io/qt-6/qchronotimer.html
> For [Qt::PreciseTimer](https://doc.qt.io/qt-6/qt.html#TimerType-enum), QChronoTimer will try to keep the precision at 1ns.

Reply:
I fear this much more complex, you must tell the operating systems, that the system wide event scheduler must not throttle down for energy savings.
This is not possible with Qt commands to my knowledge. This requires native OS commands like https://learn.microsoft.com/en-us/windows/win32/api/timeapi/nf-timeapi-timebeginperiod on Windows (which I tried btw., and didn't work for me).
On my system (I guess it's the same on all Windows 11 systems) the minimum period is 15.5ms for example.

Reply:
HID input reports are buffered in a FIFO running inside the operating systems kernel. Normally, we process them rather fast and there are seldom more than two reports in this buffer. It's safe to assume, that reports are never skipped, due to buffer overload.
If we assume, that there is a realtime timer in the controller, we can use the average time between two reports to determine the time, when the report was sent.
Please note, that sending HID output reports can take much longer than receiving input reports.